Network/XmlRpc/Internals.hs

```diff
@@ -319,6 +323,13 @@ instance XmlRpcType CalendarTime where
     fromValue = liftM localTimeToCalendarTime . fromValue
     getType _ = TDateTime
 
+instance XmlRpcType () where
+    toValue = ValueNil
```

Comment:
Then here you would have `toValue = const ValueNil` here and `f ValueNil = Just ()` below.